RPCServer/rpcwallet.py

```diff
@@ -199,31 +208,54 @@ def backup_wallet(self, afilename="bwallet.zip"):
         return True
         
         
-    def dump_wallet(self):
+    def dump_wallet(self, afilename="dump.txt", version='n/a'):
         """
         Sends back a list of all privkeys for the wallet.
         """
-        
-        # TODO - Help is welcome.
-        
         """
-        Wallet is a directory, referenced by self.path
-        each account is a json file, within a directory that is 2 chars long.
-        account abcdef is abcedf.json and lies in dir "ab"
-        
-        This dump has to parse all wallet json files (including "default.json" right in .wallet)
-        For each wallet, parse the "addresses" list and build the list of all addresses.
-        (ony Bismuth addresses, not the full priv/pub list)
-        
-        Please have a test setup and make sure code works before requesting PR.
+            Test possible path existence
         """
-        
-        print("TODO: Dump Wallet")
-        
-        # Should return a list of string or raise an exception
-        return ['TODO']        
+        dump_path = os.path.dirname(os.path.abspath(afilename))
 
+        if not os.path.exists(dump_path):
+            raise InvalidPath
 
+        if not os.path.exists(afilename):
+            if self.verbose:
+                print(afilename, "does not exist, creating")
+
+        with open(afilename, 'w') as outfile:
+            """ Write basic output """
+            outfile.write('# Wallet dump created by bismuthd ' + version + '\n')
```

Comment:
I'd prefer the use of
"some string with a {}".format(var)
instead of direct string concatenation.